packages/commerce-sdk-react/src/hooks/ShopperBaskets/query.ts

```diff
@@ -1,172 +1,234 @@
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 import {ApiClients, Argument, DataType} from '../types'
+import useCommerceApi from '../useCommerceApi'
 import {useQuery} from '../useQuery'
-import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
-import {NotImplementedError} from './../utils'
 
 type Client = ApiClients['shopperBaskets']
 
-type UseBasketParameters = NonNullable<Argument<Client['getBasket']>>['parameters']
-type UseBasketHeaders = NonNullable<Argument<Client['getBasket']>>['headers']
-type UseBasketArg = {
-    headers?: UseBasketHeaders
-    rawResponse?: boolean
-} & UseBasketParameters
 /**
  * A hook for `ShopperBaskets#getBasket`.
  * Gets a basket.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=getBasket} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperbaskets.shopperbaskets-1.html#getbasket} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-function useBasket(
-    arg: Omit<UseBasketArg, 'rawResponse'> & {rawResponse?: false},
-    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
-): UseQueryResult<DataType<Client['getBasket']>, Error>
-function useBasket(
-    arg: Omit<UseBasketArg, 'rawResponse'> & {rawResponse: true},
-    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
-): UseQueryResult<Response, Error>
-function useBasket(
-    arg: UseBasketArg,
-    options?: UseQueryOptions<DataType<Client['getBasket']> | Response, Error>
-): UseQueryResult<DataType<Client['getBasket']> | Response, Error> {
-    const {headers, rawResponse, ...parameters} = arg
-    const defaultOptions = {
-        enabled: !!parameters.basketId
-    }
-    return useQuery(
-        ['/baskets', parameters.basketId, arg],
-        (_, {shopperBaskets}) => shopperBaskets.getBasket({parameters, headers}, rawResponse),
-        {...defaultOptions, ...options}
-    )
-}
+export const useBasket = (
+    apiOptions: Argument<Client['getBasket']>,
+    queryOptions: Omit<UseQueryOptions<DataType<Client['getBasket']>>, 'queryFn'> = {}
+): UseQueryResult<DataType<Client['getBasket']>> => {
+    const {shopperBaskets: client} = useCommerceApi()
+    const method = (arg: Argument<Client['getBasket']>) => client.getBasket(arg)
+    const requiredParameters = ['organizationId', 'basketId', 'siteId'] as const
+    // Parameters can be set in `apiOptions` or `client.clientConfig`; they are merged in the helper
+    // hook, so we use a callback here that receives that merged object.
+    const getQueryKey = <T extends Record<string, unknown>>(parameters: T) =>
```

Comment:
Do we have to conduct queryKey as fn to call it from outside? all the queryKey can be achieved from the arguments that are passed. 
```
const {parameters} = apiOptions
const {shortCode, version, organizationId, siteId, locale(??)} = client.clientConfig.parameters
```

Reply:
My goal was to minimize the logic in the endpoint hooks, but I guess adding a callback increases complexity more than saving a line is worth.